common/djangoapps/cache_toolbox/tests/test_middleware.py

```diff
@@ -0,0 +1,30 @@
+"""Tests for cached authentication middleware."""
+
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.test import TestCase
+from mock import patch
+
+from student.tests.factories import UserFactory
+
+
+class CachedAuthMiddlewareTestCase(TestCase):
+    """Tests for CacheBackedAuthenticationMiddleware class."""
+
+    def setUp(self):
+        super(CachedAuthMiddlewareTestCase, self).setUp()
+        password = 'test-password'
+        self.user = UserFactory(password=password)
+        self.client.login(username=self.user.username, password=password)
+
+    def test_change_session_hash(self):
+        """
+        Verify that if a user's session auth hash and the request's hash
+        differ, the user is logged out.
+        """
+        dashboard_url = reverse('dashboard')
```

Comment:
`'/'` redirects to the dashboard on LMS, but redirects to `/home` in Studio, so we'd need to assert that `response` redirects to the right thing in a non-general way. I think I'll just make an LMS-specific and CMS-specific version of the test.